test/functional/services/filter_bar.js

```diff
@@ -44,18 +45,24 @@ export function FilterBarProvider({ getService }) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} removeFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2001);
     }
 
     async toggleFilterEnabled(key) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} disableFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2002);
     }
 
     async toggleFilterPinned(key) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} pinFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2003);
```

Comment:
The tooltip and aria-label on the button will change, we could look at those or add a data-test-subj. This does not cause a search reload so I'm surprised it would cause a race condition to occur.

Reply:
You're right that pinning/unpinning shouldn't cause a reload.  I just added the sleep in this case to be consistent across the actions.  I'll just remove this sleep and test a few times.